File: python/j21.arguments.py
#!/usr/bin/python
import sys

# illustrative variables 
ARG_COUNT = len(sys.argv) - 1 # get num of real arguments

print "The arugments passed are:"
# use collection loop with range to enumerate arguments
#   Note: range to generate indexes of argument positions
for count in range(ARG_COUNT,0,-1): # range to gen sequence of nums
   arg = sys.argv[count]                    # get arg using index
   # print count and argument using count index
   print " item %d: %s" % (count,arg) 

File: perl/j21.arguments.pl
#!/usr/bin/perl
# illustrative variables
my $last  = $#ARGV; # get index of last element
my $first = 0;      # set index of first element 

print "The arguments passed are (reverse order):\n";
# collection loop with range to enumerate args
#   Notes:
#    - $count     is index into array
#    - $count + 1 is count that is displayed to use
foreach $count (reverse($first..$last)) {
  my $arg = $ARGV[$count];  # get arg using index
  # output count and argument using count index
  printf "  item %d: %s\n", $count+1, $arg;
}